### PR TITLE
General fixes

### PR DESCRIPTION
**Pull request by SixtyFiveOhTwo**

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -115,16 +116,15 @@ public RobotContainer() {
...
-
         m_BallReciever.start();
         m_limelightReciever.start();
+
+        thumbLeft.whenPressed(new PistonInCmd(m_intake));
+        thumbRight.whenPressed(new PistonOutCmd(m_intake));
+        rightShoulder.whileHeld(new OuttakeMotorTest(m_intake));
+        yButton.whileHeld(new teleopIndex(index));
+        bButton.whileHeld(new IntakeMotorTest(m_intake));
+
```
Comment (zdonal1573) [inline src/main/java/frc/robot/RobotContainer.java]: Where is the shooter/flywheel button

File: src/main/java/frc/robot/commands/OuttakeMotorTest.java [hunk 2]
```diff
@@ -3,21 +3,23 @@
...
 package frc.robot.commands;
+
 import frc.robot.subsystems.IntakeSubsystem;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 
 public class OuttakeMotorTest extends CommandBase {
-  IntakeSubsystem Intake = new IntakeSubsystem();
+  IntakeSubsystem Intake;
   /** Creates a new IntakeMotorTest. */
   public OuttakeMotorTest(IntakeSubsystem subsystem) {
```
Comment (zdonal1573) [inline src/main/java/frc/robot/commands/OuttakeMotorTest.java:13]: Whats the difference between the intake and the outtake, this file should at least have differently named stuff probably?

File: src/main/java/frc/robot/subsystems/CanalSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+/**
+ * Manages the canal subsystem, runs the front and back belts.
+ */
+public class CanalSubsystem extends SubsystemBase {
+  /** Creates a new CanalSubsystem. */
```
Comment (zdonal1573) [inline src/main/java/frc/robot/subsystems/CanalSubsystem.java:22]: is there supposed to be nothing in the canal subsystem?

File: src/main/java/frc/robot/subsystems/Index.java [hunk 4]
```diff
@@ -3,17 +3,19 @@
...
 public class Index extends SubsystemBase {
     SparkMaxPIDController PID;
     CANSparkMax indexer;
 
-    public Index(){
+    public Index() {
 
-        indexer = new CANSparkMax(16, CANSparkMaxLowLevel.MotorType.kBrushless);
+        indexer = new CANSparkMax(Constants.BACK_CANAL_ID, CANSparkMaxLowLevel.MotorType.kBrushless);
         PID = indexer.getPIDController();
```
Comment (zdonal1573) [inline src/main/java/frc/robot/subsystems/Index.java]: This is my messed up Limelight index for the other robot :D. This whole subsystem should be deleted and we use the other intake/index subsystems

**Review by zdonal1573 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -138,7 +138,7 @@ public RobotContainer() {
     private void configureButtonBindings() {
         drivetrain.setDefaultCommand(new teleopDrive(drivetrain, () -> joystick.getRawAxis(Constants.LEFT_AXIS),
                 () -> joystick.getRawAxis(Constants.RIGHT_AXIS)));
-        cButton.whenPressed(new zeroHeading(drivetrain));
+        xButton.whenPressed(new zeroHeading(drivetrain));
         thumbLeft.whenPressed(new PistonInCmd(m_intake));
         thumbRight.whenPressed(new PistonOutCmd(m_intake));
```
Comment (zdonal1573) [inline src/main/java/frc/robot/RobotContainer.java:138]: Why are buttons in both Configure Button Bindings and Robot Container?

**Review by zdonal1573 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/CanalSubsystem.java (hunk 3 above)
Comment (KevinH45) [reply src/main/java/frc/robot/subsystems/CanalSubsystem.java:22]: I don't know why, but the version of Canal Subsystem that actually has stuff on it was never on the repo. I'm pushing it now.

File: src/main/java/frc/robot/commands/teleopCanal.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    // canal.setSpeedBack(-Constants.BELT_SPEED);
```
Comment (KevinH45) [inline src/main/java/frc/robot/commands/teleopCanal.java]: Why are all the canal functions commented out?

Comment (zdonal1573) [reply src/main/java/frc/robot/commands/teleopCanal.java]: "angry" -ryan

File: src/main/java/frc/robot/subsystems/CanalSubsystem.java (hunk 3 above)
Comment (KevinH45) [reply src/main/java/frc/robot/subsystems/CanalSubsystem.java:22]: Fixed

**Review by Spenceroni-Pepperoni — APPROVED** [review]
Looks good! BCSH need to reorganize

**Review by Gliu1211 — APPROVED** [review]
Looks good

**Review by KevinH45 — APPROVED** [review]
Looks good to me.

**Review by zdonal1573 — APPROVED** [review]

**Pull request merged**